### ISSUE TITLE
[miranda-ng.org website] HTTPS should be enabled for all miranda domains

### ISSUE THREAD

**Issue by LittleVulpix** [miranda-ng/miranda-ng#697]
Self explanatory.

Especially for the forum where user has to login using username and password.

Starting with Chrome 56, chrome will warn a user that they are entering a password on an insecure page:
https://developers.google.com/web/updates/2016/10/avoid-not-secure-warn

Starting with Firefox 51, Firefox will do the same:
https://support.mozilla.org/en-US/kb/insecure-password-warning-firefox
https://www.mozilla.org/en-US/firefox/51.0/releasenotes/

Recommended method of approach is to set up let's encrypt certificate. ( https://letsencrypt.org/ ).

To verify the level of security, I recommend https://www.ssllabs.com/ssltest/ . Rating of at least A (but preferably A+ ) should be achieved.

**Comment by LittleVulpix** [comment]
Additional inspiration from #374 , but obviously this would also mean pluginupdater would have to download miranda binaries from https too. This is however implied already by switching to https, so no new ticket needs to be logged for it.

**Comment by oodbur** [comment]
please consider publishing secure hash values for downloads afterwards.

**Comment by dartraiden** [comment]
- [ ] www.miranda-ng.org
- [ ] vi.miranda-ng.org
- [x] forum.miranda-ng.org
- [ ] wiki.miranda-ng.org
- [x] PluginUpdater
- [ ] CrashDumper

**Comment by LittleVulpix** [comment]
Please consider hardening the https parameters:

https://www.ssllabs.com/ssltest/analyze.html?d=forum.miranda-ng.org

**Comment by Robyer** [comment]
> When I try come to www and vi by http**s**, browser redirect me to plain http.
I need to update it in code.. later.

**Comment by Robyer** [comment]
> When I try come to www and vi by https, browser redirect me to plain http.

@dartraiden Done. Now it always redirects to https. Also links to other subdomains should be https too.

**Comment by Robyer** [comment]
And commited change of all http://miranda-ng strings to https://miranda-ng

Is something still remaining now?

**Comment by tweimer** [comment]
@Robyer Stylesheets in Wiki are loaded from non-secure sources, please check with Google Chrome:
![wiki](https://user-images.githubusercontent.com/2698843/29001547-ae2a8dda-7a8d-11e7-8912-462e0674ba39.png)

**Comment by tweimer** [comment]
Also links to supported Protocols on main page are still http:

**Comment by Robyer** [comment]
@tweimer Hmm, so who is maintainer of the wiki? Is he aware of that?
I'll fix the protocols on the main site, thanks.

**Comment by tweimer** [comment]
@Robyer I thought you are the maintainer? Maybe @watcherhd was. Not sure who is.

**Comment by Robyer** [comment]
@tweimer I maintain main site, but that's all. I don't know who does forum and wiki. Perhaps @kxepal ?

**Comment by Robyer** [comment]
Hmm, I can't change links to wiki to https until the wiki itself is fixed...

**Comment by dartraiden** [comment]
@Robyer, we also can replace strings like "AdvaImg plugin not found. Please get it from http://miranda-ng.org/p/AdvaImg/ to be able to use images." to https, because wiki automatically converts https://miranda-ng.org/p/AdvaImg/ to http://wiki.miranda-ng.org/index.php?title=Plugin:AdvaImg

**Comment by kxepal** [comment]
We're awaiting  Let's Encrypt wildcard domain support which announced to happen at the end of this year. After that we'll move all the subdomains to HTTPS, be sute (:

**Comment by kxepal** [comment]
Hm...It @georgehazan setup Let's Encrypt everywhere. I missed that point. Ok, so we have wiki broken. That's fixable.

**Comment by dartraiden** [comment]
@kxepal, and problem with "mixed content" on forum.

**Comment by kxepal** [comment]
[wiki](https://wiki.miranda-ng.org/) is secured now.

**Comment by kxepal** [comment]
@dartraiden fixed.

**Comment by Robyer** [comment]
@kxepal Perfect. Can you also force wiki to redirect from http:// to https:// automatically? 

@dartraiden I already changed the strings in plugins, see my latest commits. I will in few minutes commit fixes to wiki links too, so it will be complete.

**Comment by Robyer** [comment]
Ok, now it's everywhere in website's sources and in Miranda plugins' sources changed to https :-)

**Comment by dartraiden** [comment]
https://github.com/miranda-ng/miranda-ng/blob/master/include/delphi/m_crypto.inc
https://github.com/miranda-ng/miranda-ng/blob/master/plugins/ZeroNotification/src/stdafx.h (and many others stdafx.h)

**Comment by Robyer** [comment]
@dartraiden Right, I forgot that there exists other files than *.cpp and *.h :-) Now I checked all files, so it should be finally all now.

**Comment by dartraiden** [comment]
@Robyer, one link on https://miranda-ng.uservoice.com :)
And some Chinese spam...

**Comment by LittleVulpix** [comment]
To me it feels like uservoice is useless. Github can fill that role too :p I always forget the uservoice page even exists.

**Comment by Robyer** [comment]
@dartraiden I couldn't find where it can be changed on uservoice (if it can be changed). Spam deleted, thanks for notice.

@LittleVulpix I like it, it's role is different. It should be like Github is for issues, UserVoice is for feature requests (because other users can vote what is most important for them).

**Comment by Robyer** [comment]
I believe everything is done now and this issue could be closed?

LittleVulpix mentioned "Please consider hardening the https parameters: https://www.ssllabs.com/ssltest/analyze.html?d=forum.miranda-ng.org " Can @kxepal do that? :)

**Comment by kxepal** [comment]
@Robyer 
Oh, I missed it. Nice. Will do it.

**Comment by Robyer** [comment]
@kxepal also see this please, https://forum.miranda-ng.org/index.php?topic=4967.msg20093#msg20093

**Comment by LittleVulpix** [comment]
@Robyer @dartraiden I guess nobody set up proper autorenewal of certs? It seems to have expired today and I cannot open the page unless I accept the outdated certificate.

**Comment by georgehazan** [comment]
I did, it should work

**Comment by LittleVulpix** [comment]
Yep, it's good now:

Begin: October 9, 2017 at 5:52:01 PM
(October 9, 2017 at 3:52:01 PM GMT)

End:
January 7, 2018 at 4:52:01 PM
(January 7, 2018 at 3:52:01 PM GMT)

You can call the renewal every day, it'll simply renew when it's time for it to do so; not sure how it's being done now.

**Comment by georgehazan** [comment]
yes, that's precisely what I've done

**Comment by LittleVulpix** [comment]
Good! :)

**Comment by dartraiden** [comment]
Our https-certificate don't cover https://oauth.miranda-ng.org/

**Comment by georgehazan** [comment]
https://oauth.miranda-ng.org/ fixed
